### DISCUSSION TITLE
Shelly Plus 2PM (ESP32) Window Covering Script

### DISCUSSION THREAD

**Discussion by shanghaix** [RavenSystem/esp-homekit-devices#2062] [MEPLHAA Scripts]
Hello Everyone,

Has anyone managed to adapt the Shelly 2.5 window covering script from the device database for the new Shelly Plus 2PM?

I used to have many Shelly 2.5 devices but a lot off them have failed due to faulty capacitors and I wanted to replace them with the Shelly Plus 2PM, however I am struggling to make the script work.

This is the Shelly 2.5 script:
Window covering with external push buttons
Opening 18 seconds, closing 14 seconds, non-lineal correction 70%

{"c":{"io":[[[4,15,0],2],[[2],6],[[5,13],6,0,1]],"l":0,"b":[[2,5]]},"a":[{"t":45,"o":18,"c":14,"f":70,"0":{"r":[[15],[4,1]]},"1":{"r":[[15,1],[4]]},"2":{"r":[[15],[4]]},"3":{"r":[[15],[4,0,0.2]]},"4":{"r":[[15,0,0.2],[4]]},"f3":[[5,0]],"f4":[[13,0]]}]}

It would be awesome if someone has figured it out already!

**Comment by ceg0117** [comment]
It seems that the time to open is 3 seconds longer than the time to close Nonlinear correction seems to be of little use For reference

**Comment by shanghaix** [comment]
I have figured it out, if anyone is interested this is the script:
Make sure to use ESP32 1 core firmware (fullhaaboot_esp32_1.bin) for the Shelly Plus 2PM 
Open and closing time: 23 seconds.

{"c":{"io":[[[12,13,0],2],[[4],6],[[5,18],6,0,1]],"l":0,"b":[[4,5]]},"a":[{"t":45,"o":23,"c":23,"f":80,"0":{"r":[[13],[12,1]]},"1":{"r":[[13,1],[12]]},"2":{"r":[[13],[12]]},"3":{"r":[[13],[12,0,0.2]]},"4":{"r":[[13,0,0.2],[12]]},"f0":[[18,0]],"f1":[[5,0]],"f2":[[18],[5]]}]}

**Reply by pongista** [reply]
Sorry Shanghaix, can you help please???? I need to found a script for HAA v12 for a shelly 1 that have opening time of 5 second, after this 5 second I need to see icon state closed, only a click to open, no need anything more, should work like a lock mechanism but with a garage door Icon, please someone can help me with script???!! Thanks a lot

**Comment by CincoD2** [comment]
Is there any way to setup a single external button that opens and close roller shutter alternatively? 
It'd work with one push to open, one push to stop and one push to close.
Thank you!
